src/NLog/Internal/FileInfoExt.cs

```diff
@@ -55,7 +55,20 @@ public static DateTime GetCreationTimeUtc(this FileInfo fileInfo)
 #endif
         }
 
-       
+        public static DateTime? ValidateFileCreationTime<T>(T fileInfo, Func<T, DateTime?> primary, Func<T, DateTime?> fallback, Func<T, DateTime?> finalFallback = null)
```

Comment:
The idea of the ext classes is to write c# extensions in it ;)

Reply:
Refactored the code, so it now belongs :)

Reply:
`LookupValidFileCreationTimeUtc<T>` is still no extension? 

Also am a bit confused by `T fileinfo`. How is T a fileinfo in the type system - what am I missing?

Reply:
I think a helper is in this case better, as you should not make an extension on T here

Pity of the UnixFileInfo, but could live with that when this code is in a helper class.

Reply:
So remove extensions, and change into helpers, or have both ?

Reply:
Now introduced a static FileInfoHelper-class